src/main/js/ephox/imagetools/transformations/ImageResizerWebgl.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * ImageResizerWebgl.js
+ *
+ * Released under LGPL License.
+ * Copyright (c) 1999-2015 Ephox Corp. All rights reserved
+ *
+ * License: http://www.tinymce.com/license
+ * Contributing: http://www.tinymce.com/contributing
+ */
+
+/**
+ * Resizes image/canvas using canvas and Webgl
+ */
+define("ephox/imagetools/transformations/ImageResizerWebgl", [
+    "ephox/imagetools/util/Promise",
+    "ephox/imagetools/util/Conversions",
+    "ephox/imagetools/util/Canvas",
+    "ephox/imagetools/util/ImageSize"
+], function(Promise, Conversions, Canvas, ImageSize) {
+
+    /**
+     * @method scale
+     * @static
+     * @param image {Image|Canvas}
+     * @param dW {Number} Width that the image should be scaled to
+     * @param dH {Number} Height that the image should be scaled to
+     * @returns {Promise}
+     */
+    function scale(image, dW, dH) {
+        return new Promise(function(resolve, reject) {
+            var sW = ImageSize.getWidth(image);
+            var sH = ImageSize.getHeight(image);
+            var wRatio = dW / sW;
+            var hRatio = dH / sH;
+
+            var canvas = Canvas.create(dW, dH);
+
+            try {
+                _drawImage(canvas, image, wRatio, hRatio);
+            } catch(ex) {
+                reject(ex);
+                return;
+            }
+
+            if (image.src) {
+                Conversions.revokeImageUrl(image);
+            }
+            image = null; // just in case
+
+            resolve(canvas);
+        });
+    }
+
+    var shaders = {
+        bilinear: {
+            VERTEX_SHADER: '\
```

Comment:
"probably" is not enough. Sadly for legal reasons we need a clear statement as to whether the initialisation logic was created completely by you, or copied/inspired from an external source (and if so what source).